File: 24/swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode head2 = new ListNode();
        ListNode node = head;
        ListNode node2 = head2;
        
        while(node!=null && node.next!=null) {
            ListNode n1 = new ListNode(node.val);
            ListNode n2 = new ListNode(node.next.val);
            n2.next = n1;
            node2.next = n2;
            node = node.next.next;
            node2 = node2.next.next;
        }
        if(node!=null) {
            node2.next = new ListNode(node.val);
        }
        return head2.next;
    }
}

File: 24/swap-nodes-in-pairs.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapPairs(struct ListNode* head){
    if(head==NULL || head->next==NULL) return head;
    struct ListNode* p1 = head;
    struct ListNode* p2 = head->next;
    while(p1!=NULL && p2!=NULL){
        int temp=p1->val;
        p1->val = p2->val;
        p2->val = temp;

        if(p1->next!=NULL) p1 = p1->next->next;
        else break;
        if(p2->next!=NULL) p2 = p2->next->next;
    }
    return head;
}
